### DISCUSSION TITLE
Qual o uso correto da tag <main>?

### DISCUSSION THREAD

**Discussion by felipefialho** [frontendbr/forum#30] [Discussão]
Trazendo a discussão da issue #23 para cá.

:)

**Comment by afonsopacifer** [comment]
Segundo a descrição da [MDN](https://developer.mozilla.org/pt-BR/docs/Web/HTML/Element/main) que eu mesmo cheguei a traduzir na época...

<hr>

O elemento `<main>` define o conteúdo principal dentro do `<body>` em seu documento ou aplicação. Entende-se como conteúdo principal aquele relacionado diretamente com o tópico central da página ou com a funcionalidade central da aplicação. O mesmo deverá ser único na página, ou seja, dentro do elemento `<main>` não deverão ser incluidas seções da página que sejam comuns a todo o site ou aplicação, tais como mecanismos de navegação, informações de copyright, logotipo e campos de busca (a não ser, é claro,  caso a função principal do documento seja  fazer algum tipo de busca).

**Observações importantes**
- O `<main>` não pode ser filho dos elementos `<article>`, `<aside>`, `<footer>`, `<header>`, ou `<nav>`.
- Não se deve incluir mais de um elemento `<main>` no mesmo documento.
- O `<main>` pode ser filho de uma ou mais divs. (Apenas divs ;))

**Comment by vitorbritto** [comment]
Na _issue_ #23 eu mencionei o uso da tag `<main>` dentro da `<section>`, mas me equivoquei. Eu utilizo logo depois do `header`, para cada página.

Exemplo:

```
<header></header>
<main>
  <article>
    <header></header>
    <div></div>
    <footer></footer>
  </article>
</main>
<footer></footer>
```

**Comment by felipefialho** [comment]
Exatamente @vitorbritto, faço o mesmo uso que você.

**Comment by rickbenetti** [comment]
+1 @vitorbritto

**Comment by eduardojmatos** [comment]
@vitorbritto @LFeh @rkb81 boa, eu geralmente englobo todo mundo na main, mas faz sentido essa abordagem, do `header` ficar fora... Vou começar a usar assim =D

**Comment by rickbenetti** [comment]
Incrementando eu normalmente faço isso:

```
<header>
<div class="limit"></div>
</header>
<main>
<div class="limit">
  <article>
    <header></header>
    <div></div>
    <footer></footer>
  </article>
</div>
</main>
<footer>
<div class="limit"></div>
</footer>
```

Normalmente utilizo uma div.limit para centralizar elementos conforme a necessidade.

Quem me mostrou esta forma de codificar foi o @diegoeis quando trabalhei na visie.

**Reply by vazdiego** [reply]
Pergunta de quem é newbie ainda, esses div.limit não interferem na semântica ?

obs. começando agora em 2022.

**Comment by diegoeis** [comment]
[http://tableless.com.br/a-tag-main/](http://tableless.com.br/a-tag-main/)
[https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main)

**Comment by eduardojmatos** [comment]
@rkb81 @diegoeis boa! Agora sim, fechou de vez a questão :smile:

**Comment by dionarodrigues** [comment]
Legal @diegoeis . Esse link contém ótimas explicações.

No entanto, trazendo uma parte da discussão da [issue #23](https://github.com/frontendbr/forum/issues/23), uma das questões é justamente a que o @vitorbritto levantou: se pode ou não utilizar ela dentro de uma section.

---

Sendo assim, trago a minha explicação para essa issue:

Lendo a [documentação da W3C](https://www.w3.org/TR/html5/grouping-content.html#the-main-element) eu acho que o uso da `<main>` dentro da `<section>` pode depender do contexto. Lá também não cita explicitamente que é errado - vejam essa citação:

> Authors must not include the main element as a descendant of an article, aside, footer, header or nav element.

O mesmo que o @fdaciuk  citou com a documentação da [mozilla](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main).

No entanto, diz que `<main>` não é aconselhável para identificar o conteúdo principal de uma _sectioning content_ (e isso já envolve a `<section>`).

Então, penso que depende do contexto, pois, se a `<section>` está sendo utilizada para fins muito genéricos (algo mais amplo), entendo que se pode usar uma `<main>` dentro dela, desde que `<main>` contenha como filho uma _sectioning content_, mesmo que seja a própria `<section>`.

Acho essa tag muito importante, já que ela pode substituir a "role=main" futuramente.

**Comment by dionarodrigues** [comment]
Por exemplo, pensando que uma página contenha sidebar, o código poderá ser:

```
body 
     header
          .logo
          nav
     section
          aside
          main
               article (ou section)
                    conteúdo principal
     footer

```

Deste modo, ela não precisa necessariamente estar no mesmo nível que `<header>` e `<footer>`.

**Comment by diegoeis** [comment]
Hoje é muito difícil dizer o que é certo ou o que é errado. Tudo depende de contexto. Veja:
`section` serve para separar um conteúdo sobre um determinado assunto.
`main` serve para indicar o conteúdo principal ou a área principal do site. 

Faz mais sentido a `section` (ou várias) dentro de uma `main` e não o contrário.
Mas se você usar um `main` dentro de uma `section` você está explicitando que aquele pedaço é a área mais importante e principal do site. É ali que o Google tem que relevar, é ali que o Leitor de tela tem que priorizar e etc etc etc.

Mas, sei lá... Nesse governo Dilma, essas coisas todas podem estar erradas. hehehehe

**Comment by dionarodrigues** [comment]
HAuhahuahua Vdd

**Comment by vitorbritto** [comment]
Verdade @diegoeis. 

IMHO, cada bloco de conteúdo traçado no markup possui uma relevância e deve ser pensado previamente. 

Costumo me guiar pelo PSD ou Sketch do projeto e escrevo/digito em um papel/editor as tags representadas por  uma lista de tópicos aninhados. Com isso, eu consigo representar as tags e sua respectiva hierarquia de maneira legível, semântica e sem passar por algum retrocesso mais a frente.

Até criei uma [ferramenta com Shell Script](https://github.com/vitorbritto/markupify) para ajudar nesse processo. Assim, eu já consigo organizar o _markup_ pensando também na próxima camada (CSS).

**Comment by eduardojmatos** [comment]
@diegoeis @diogorodrigues eu acredito muito nisso do contexto. Mas ainda torço o nariz com a ideia de um `<main/>` dentro de outra tag qualquer, a não ser `html` e `body`... Mas, é contexto, e aí, vem o famoso **`depende`**.

**Comment by eduardojmatos** [comment]
@vitorbritto massa essa ferramenta :clap:

**Comment by RafaSantana** [comment]
Olá a todos, ainda não sou desenvolvedor, mas acredito que a explicação do @diegoeis faz bastante sentido. O _main_ cobrindo todo o conteúdo principal invés de se colocado dentro de uma _section_ até porque o conteúdo principal de uma pagina é apena um, o resto são facilidade de uso para interação com o usuário. Provavelmente estou falando besteira, mas pelo que pesquisei o que o @diegoeis faz bastante sentido.

**Comment by jbdevelop** [comment]
É antigo o post, mas... o **main** deve representar o conteúdo principal da página e ser único, ou seja, não se deve repetí-lo na página. Não deve estar dentro de **article**, **section**, etc. Está na documentação. Creio que um bom uso seria...

```
<body> 
<header></header>
<main></main>
<footer></footer>
</body>
```
.....

https://developer.mozilla.org/pt-BR/docs/Web/HTML/Element/main

**Reply by AllysonGerald** [reply]
> É antigo o post, mas... o **main** deve representar o conteúdo principal da página e ser único, ou seja, não se deve repetí-lo na página. Não deve estar dentro de **article**, **section**, etc. Está na documentação. Creio que um bom uso seria...
> 
> ```
> <body> 
> <header></header>
> <main></main>
> <footer></footer>
> </body>
> ```
> 
> .....
> 
> https://developer.mozilla.org/pt-BR/docs/Web/HTML/Element/main

Simples e lógico.

**Comment by italosilva01-zz** [comment]
Acredito que um bom exemplo para usar a tag main é em um site de notícias. Onde a notícia mais relevante do dia fica dentro de uma main e as demais e section fora da main
